### ISSUE TITLE
WebXR not working on Microsoft Mixed Reality 

### ISSUE THREAD

**Issue by osfog** [potree/potree#994]
Hi,

Potree does not seem work for use in Microsoft Mixed Reality

In firefox and Chrome with the WebXR API Emulator from Mozilla it seems to work ok. 

However when running on windows with their Edge browser (or Chrome) the scenes do not work at all. The transfer to the HP Reverb G2 initiates but then you are left with a dialog in the VR display:

`"The page isn´t responding"`

Looking into the console on edge you get the error: 
```
Uncaught (in promise) DOMException: Failed to construct 'XRWebGLLayer': 
WebGL context must be marked as XR compatible in order to use with an immersive XRSession
````

Addressing this by introducing the fix listed here: 
[https://github.com/mrdoob/three.js/issues/20715](https://github.com/mrdoob/three.js/issues/20715)
removes the XRWebGLLayer error but does little else - there is still no scene loaded

The pure three.js demos works fine. 

The behavour seems to be the same for all potrees one example is: 
https://potree.org/potree/examples/vr_candi_banyunibo.html

Would be grateful for some hints on what to can be stripped out in order to create a barebone test or other ideas on how this best can be addressed.

```
Microsoft Edge
Version 89.0.774.45 (Official build) (64-bit)


Google Chrome 
Version 89.0.4389.82 (Official Build) (64-bit)
```

**Comment by osfog** [comment]
I can add that even a scene without any pointcloud loaded at all the same issue is there ( page not responding in when switching to the display)

**Comment by osfog** [comment]
For completeness i managed to get it up and running also on Microsoft Mixed Reality -  What I have done was to step three.js, removed some of the controller related functionality now it works if you enter (same issue as before) exit VR and then enter again....  i actually have seen this issue also on other browsers.  Not sure what of the steps had beneficial impact  - it might have been just to enter VR the second time...

**Comment by bbirbo** [comment]
The same issue is happening with HTC Vive and Steam VR.
the examples here [https://immersive-web.github.io/webxr-samples/](https://immersive-web.github.io/webxr-samples/) are working great though.

Can someone take a look please.

**Comment by osfog** [comment]
yep and perhaps even more relevant the https://threejs.org/examples/webxr_vr_cubes.html seems also to work fine. To me it looks like there is something in layer on top of three.js in potree that is causing the issue - as it, at least in some cases, is intermittent  (initiating the scene works after having entered and exited VR mode once) it in my eyes hints an issue during the initiation or state management.

**Comment by bbirbo** [comment]
Potree is the best pointcloud viewer out there period.
The VR feature is the cherry on top and we have been waiting so long for this. It is very frustrating though that it is not working even when the implementation is there. I am hoping someone will fix this issue soon.
Thanks.

**Comment by mvigerske** [comment]
I tried around for quite a while and suddenly it worked. Then it didn't work again. And then I found out: when I switch to VR mode on the PC in Chrome with "Enter VR", I get the aforementioned error message. If I then take off the glasses (Oculus Rift or Oculus Quest with Link cable) and put them back on and tap "Enter VR" again, everything works. This can be reproduced. Every time I take the glasses off and restart the VR mode, it works.

With the Quest (without cable) (and presumably with HTC) there is the app Firefox Reality. With it, I can call Potree directly in the glasses and the VR mode starts directly, without an error message.

**Comment by Cornelius000** [comment]
Seems to be related to [this](https://github.com/mrdoob/three.js/issues/21126). Maybe an update of threejs will do, if thats possible

**Comment by Karl-Keller** [comment]
I'm running my own web server for Potree and it's completely functional for normal clouds.  However, using a Quest system and browser none of the example VR files from Potree load.  The potree.org VR file links, e.g. https://potree.org/potree/examples/vr_heidentor.html, work correctly.  Any thoughts about where I'm going wrong much appreciated.

**Comment by mvigerske** [comment]
Hi Karl, I made a note of two things that I must not activate in order for a VR view to work. Once EDL must not be on: "viewer.setEDLEnabled(false);" and I must not use "PointShape.PARABOLOID" for the point clouds.
